docs/rfcs/rfc-035-authorization.md

```diff
@@ -0,0 +1,759 @@
+    authorConnection_NOT: UserAuthorConnectionWhere
+    author_NOT: UserWhere
+    id: ID
+    id_CONTAINS: ID
+    id_ENDS_WITH: ID
+    id_IN: [ID!]
+    id_NOT: ID @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    id_NOT_CONTAINS: ID @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    id_NOT_ENDS_WITH: ID @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    id_NOT_IN: [ID!] @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    id_NOT_STARTS_WITH: ID @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    id_STARTS_WITH: ID
+    name: String
+    name_CONTAINS: String
+    name_ENDS_WITH: String
+    name_IN: [String!]
+    name_NOT: String @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    name_NOT_CONTAINS: String @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    name_NOT_ENDS_WITH: String @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    name_NOT_IN: [String!] @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    name_NOT_STARTS_WITH: String @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    name_STARTS_WITH: String
+}
+```
+
+#### JWT Payload schema
+
+The generated types for the directive will depends on `jwtPayload` when instantiating the feature. Given the following:
+
+```ts
+new Neo4jGraphQL({
+  typeDefs,
+  features: {
+    authorization: {
+      secret: "secret",
+      jwtPayload: {
+        type: "object",
+        properties: {
+            sub: { type: "string" },
+            roles: {
+                type: "array",
+                items: { type: "string" }
+            }
+        }
+      }
+    }
+  }
+})
+```
+
+For the purposes of filtering, this will generate the following input type:
+
+```gql
+input JWTPayloadWhere {
+    AND: [JWTPayloadWhere!]
+    NOT: JWTPayloadWhere
+    OR: [JWTPayloadWhere!]
+    sub: String
+    sub_CONTAINS: String
+    sub_ENDS_WITH: String
+    sub_IN: [String!]
+    sub_STARTS_WITH: String
+    roles: [String!]
+    roles_INCLUDES: String
+}
+```
+
+### Authorization directive definitions
+
+Firstly, for `User`:
+
+```gql
+input UserAuthorizationWhere {
+  OR: [UserAuthorizationWhere!]
+  AND: [UserAuthorizationWhere!]
+  NOT: UserAuthorizationWhere
+  jwtPayload: JWTPayloadWhere
+  node: UserWhere
+}
+
+enum AuthorizationFilterOperation {
+  READ
+  UPDATE
+  DELETE
+  CREATE_RELATIONSHIP
```

Comment:
For the same reason we're leaning towards `CONNECT`/`DISCONNECT` for the condition schema generation, should it also not be `CONNECT`/`DISCONNECT` here? This is what users currently understand our operations as. 

I do agree we should move in this direction in the future but I don't know if we're there yet

Reply:
I was actually about to rewrite the read/write controls RFC with `CREATE_RELATIONSHIP` and `DELETE_RELATIONSHIP`, as this will align the terminology across that, authorization, and Subscriptions, and future-proof us so we don't have to make more breaking changes in the future.

Reply:
As long as we're going to be aligned in multiple places then I think this is fine then